megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -1481,7 +1481,9 @@ private static String toHitIsImpossible(IGame game, Entity ae, int attackerId, T
         // for spheroid dropships in atmosphere (and on ground), the rules about
         // firing arcs are more complicated
         // TW errata 2.1
-        if (Compute.useSpheroidAtmosphere(game, ae) && weapon != null) {
+        if ((Compute.useSpheroidAtmosphere(game, ae) ||
+                (((IAero) ae).isSpheroid() && ae.getAltitude() == 0))
```

Comment:
I don't see anything that guarantees the Entity is an aerospace unit at this point, which means that this can throw a ClassCastException.